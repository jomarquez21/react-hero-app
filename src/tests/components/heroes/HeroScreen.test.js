import React from 'react';
import {MemoryRouter, Route} from 'react-router-dom';
import {mount} from 'enzyme';
import {HeroScreen} from './../../../components/heroes/HeroScreen';
import '@testing-library/jest-dom';

describe('Test on `HeroScreen`', () => {
  const historyMock = {
    push: jest.fn(),
    length: 10,
    goBack: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  })
  
  test('Deberia renderizar el componente redirect si no hay argumentos en el URL', () => {
    const component = mount(
      <MemoryRouter initialEntries={['/hero']}>
        <HeroScreen history={historyMock} />
      </MemoryRouter>
    );

    expect(component.find('Redirect').exists()).toBe(true);
  });

  test('Deberia renderizar un hero', () => {
    const component = mount(
      <MemoryRouter initialEntries={['/hero/marvel-captain']}>
        <Route path="/hero/:heroeId" component={HeroScreen} />
      </MemoryRouter>
    );

    expect(component).toMatchSnapshot();
    expect(component.find('.row').exists()).toBe(true);
  });

  test('Debe regresar a la pantalla anterior con push', () => {
    const component = mount(
      <MemoryRouter initialEntries={['/hero/marvel-captain']}>
        <Route
          path="/hero/:heroeId"
          component={() => <HeroScreen history={{...historyMock, length: 1}} />}
        />
      </MemoryRouter>
    );

    component.find('button').prop('onClick')();
    expect(historyMock.push).toHaveBeenCalledWith('/');
    expect(historyMock.push).toHaveBeenCalledTimes(1);
    expect(historyMock.goBack).not.toHaveBeenCalled();
  });

  test('Debe regresar a la pantalla anterior con push', () => {
    const component = mount(
      <MemoryRouter initialEntries={['/hero/marvel-captain']}>
        <Route
          path="/hero/:heroeId"
          component={() => <HeroScreen history={historyMock} />}
        />
      </MemoryRouter>
    );

    component.find('button').prop('onClick')();
    expect(historyMock.goBack).toHaveBeenCalledTimes(1);
    expect(historyMock.push).not.toHaveBeenCalled();
  });

  test('Debe llamar al redirect si el hero no existe', () => {
    const component = mount(
      <MemoryRouter initialEntries={['/hero/jomarquez']}>
        <Route
          path="/hero/:heroeId"
          component={() => <HeroScreen history={historyMock} />}
        />
      </MemoryRouter>
    );

    expect(component.text().trim()).toBe('');
  });
});
