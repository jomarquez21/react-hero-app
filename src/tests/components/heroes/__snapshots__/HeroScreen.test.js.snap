// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on \`HeroScreen\` Deberia renderizar un hero 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Captain America"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="../assets/heroes/marvel-captain.jpg"
    />
  </div>
  <div
    className="col-8 animate__animated animate__fadeIn"
  >
    <h3>
       
      Captain America
       
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
         
        <b>
           Alter ego: 
        </b>
         
        Steve Rogers
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
           Publisher: 
        </b>
         
        Marvel Comics
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
           First appearance: 
        </b>
         
        Captain America Comics #1
         
      </li>
    </ul>
    <h5>
       Characters 
    </h5>
    <p>
       
      Steve Rogers
       
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
